modules/openapi-generator/src/main/resources/cpp-qt5-client/api-body.mustache

```diff
@@ -226,220 +226,231 @@ void {{classname}}::{{nickname}}({{#allParams}}const {{{dataType}}} &{{paramName
+                case QJsonValue::Array:
+                {
+                    paramString.append(key+assignOperator+QVariant(parameter.value(key).toArray()).toString());
+                    break;
+                }
+                case QJsonValue::Object:
+                {
+                    paramString.append(key+assignOperator+QVariant(parameter.value(key).toObject()).toString());
+                    break;
+                }
+                case QJsonValue::Null:
+                case QJsonValue::Undefined:
+                    break;
             }
-            case QJsonValue::Null:
-            case QJsonValue::Undefined:
-                break;
+            count++;
         }
-        count++;
-    }
-    fullPath.replace({{paramName}}PathParam, QUrl::toPercentEncoding(paramString));
-    {{/isPrimitiveType}}
-    {{#isPrimitiveType}}
-    QString paramString = (pathStyle == "matrix") ? pathPrefix+"{{baseName}}"+pathSuffix : pathPrefix;
-    fullPath.replace({{paramName}}PathParam, paramString+QUrl::toPercentEncoding(::{{cppNamespace}}::toStringValue({{paramName}})));
-{{/isPrimitiveType}}{{/collectionFormat}}{{#collectionFormat}}
-    if ({{{paramName}}}.size() > 0) {
-            QString paramString = (pathStyle == "matrix") ? pathPrefix+"{{baseName}}"+pathSuffix : pathPrefix;
-            qint32 count = 0;
-            foreach ({{{baseType}}} t, {{paramName}}) {
-                if (count > 0) {
-                    fullPath.append(pathDelimiter);
+        fullPath.replace({{paramName}}PathParam, QUrl::toPercentEncoding(paramString));
+        {{/isPrimitiveType}}
+        {{#isPrimitiveType}}
+        QString paramString = (pathStyle == "matrix") ? pathPrefix+"{{baseName}}"+pathSuffix : pathPrefix;
+        fullPath.replace({{paramName}}PathParam, paramString+QUrl::toPercentEncoding(::{{cppNamespace}}::toStringValue({{paramName}}{{^required}}.value<{{{dataType}}}>(){{/required}})));
+    {{/isPrimitiveType}}{{/collectionFormat}}{{#collectionFormat}}
+        if ({{{paramName}}}{{^required}}.value<{{{dataType}}}>(){{/required}}.size() > 0) {
+                QString paramString = (pathStyle == "matrix") ? pathPrefix+"{{baseName}}"+pathSuffix : pathPrefix;
+                qint32 count = 0;
+                foreach ({{{baseType}}} t, {{paramName}}{{^required}}.value<{{{dataType}}}>(){{/required}}) {
+                    if (count > 0) {
+                        fullPath.append(pathDelimiter);
+                    }
+                    fullPath.append(QUrl::toPercentEncoding(::{{cppNamespace}}::toStringValue(t)));
+                    count++;
                 }
-                fullPath.append(QUrl::toPercentEncoding(::{{cppNamespace}}::toStringValue(t)));
-                count++;
-            }
-              fullPath.replace({{paramName}}PathParam, paramString);
-    }
-{{/collectionFormat}}{{/pathParams}}
+                fullPath.replace({{paramName}}PathParam, paramString);
+        }
+    {{/collectionFormat}}
+}
```

Comment:
indentation

Reply:
yea this was actually weirdly intentional

see that samples/client/petstore/cpp-qt5/client/PFXStoreApi.cpp and it is formatted correctly (with this indented once, it gets double indented in the output)

I will revisit it and see if there is something I am missing/a way of using what was there before to help format it right

Reply:
I was able to clean it up a bit